devices/oneplus-bacon/kernel/default.nix

```diff
@@ -0,0 +1,42 @@
+{
+  mobile-nixos
+, fetchFromGitHub
+, kernelPatches ? [] # FIXME
+, buildPackages
+, dtbTool
+}:
+
+(mobile-nixos.kernel-builder-gcc6 {
```

Comment:
Try building with `kernel-builder-gcc-49`.

Some vendor kernels *require* a specific compiler version to even boot. Generally when it doesn't boot, IIRC it will hang. Yes, it doesn't make sense.